perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1578,6 +1578,32 @@ public void updateSponsorshipValidity(PerunSession sess, Member sponsoredMember,
 		membersManagerBl.updateSponsorshipValidity(sess, sponsoredMember, sponsor, newValidity);
 	}
 
+	@Override
+	public List<RichMember> getRichMembersByIds(PerunSession sess, List<Integer> ids, List<String> attrNames) throws PrivilegeException, AttributeNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getRichMembersByIds_List<Integer>_List<String>_policy")) {
+			throw new PrivilegeException(sess, "getRichMembersByIds");
+		}
+
+		List<Member> members = getMembersManagerBl().getMembersByIds(sess, ids);
+		members.removeIf(member -> !AuthzResolver.authorizedInternal(sess, "filter-getRichMembersByIds_List<Integer>_List<String>_policy", member));
+
+		List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attributeDefinitions.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		List<RichMember> richMembers = membersManagerBl.convertMembersToRichMembers(sess, members);
+		//Enriched rich members with attributes by list of attributes
+		richMembers = membersManagerBl.convertMembersToRichMembersWithAttributes(sess, richMembers, attributeDefinitions);
+		//RichMembers with filtered attributes by rights from session
+		richMembers = membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers);
```

Comment:
There is no need to save the output of filtering to the variable. You can return the whole output of the filtering instead.

Reply:
Fixed